agora/Agora/Utils.hs

```diff
@@ -619,6 +620,14 @@ scriptHashFromAddress = phoistAcyclic $
       PScriptCredential ((pfield @"_0" #) -> h) -> pcon $ PJust h
       _ -> pcon PNothing
 
+-- | Return true if the given address is a script address.
+isScriptAddress :: Term s (PAddress :--> PBool)
```

Comment:
Can you update TreasuryWithdrawal effect to use this. Check line 143. Thanks!

Reply:
Sure thing!